src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/LSPRazorProjectHost.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+using Microsoft.VisualStudio.LanguageServer.Client;
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    [AppliesTo("(DotNetCoreRazor & DotNetCoreRazorConfiguration) | ((DotNetCoreRazor | DotNetCoreWeb) & !DotNetCoreRazorConfiguration)")]
+    [Export(ExportContractNames.Scopes.UnconfiguredProject, typeof(IProjectDynamicLoadComponent))]
+    internal class LSPRazorProjectHost : IProjectDynamicLoadComponent
+    {
+        private readonly JoinableTaskFactory _joinableTaskFactory;
+        private readonly LSPEditorFeatureDetector _featureDetector;
+        private readonly List<Lazy<ILanguageClient, ILanguageClientMetadata>> _applicableLanguageClients;
+        private readonly Lazy<ILanguageClientBroker> _languageClientBroker;
+
+        [ImportingConstructor]
+        public LSPRazorProjectHost(
+            JoinableTaskContext joinableTaskContext,
+            LSPEditorFeatureDetector featureDetector,
+            Lazy<ILanguageClientBroker> languageClientBroker,
+            [ImportMany] IEnumerable<Lazy<ILanguageClient, IDictionary<string, object>>> languageClients)
```

Comment:
>IDictionary<string, object> [](start = 59, length = 27)

Can't you specify a metadata contract here instead of the dictionary? something like:

[ImportMany] IEnumerable<Lazy<ILanguageClient, ILanguageClientMetadata>> languageClients)

which allows you to work with it strongly typed?

Reply:
Because `ILanguageClientMetadata` doesn't expose ContentTypes

Reply:
Actually it does expose it through `IContentTypeMetadata`. Hmm, I don't remember why I originally did this. Might have just been an oversight.

Reply:
The `ClientName` isn't accessible through the `IContentTypeMetadata` (it only has the `ContentTypes` property).

Looking at the LSP Platform loader, they use a `IDictionary<string, object>` as well. I believe this will need to be left as is.

Reply:
But `ILanguageClientMetadata` inherits from `IContentTypeMetadata` so they should both be accessible right? Worth a shot if it makes the code less complicated.

Reply:
I've tried updating to `ILanguageClientMetadata` but only the `RazorCsharp` language server is accepted (as opposed to htmlyLSP and all the other language servers with the current implementation).

Per my understanding, the remaining language servers don't export `ILanguageClientMetadata` and thus aren't being provided by MEF. Consequently, we'll have to leave our implementation here as is (and in line with what LSP Platform does) - confirmed with @ajaybhargavb offline, thanks!